pyvista/core/utilities/geometric_sources.py

```diff
@@ -498,10 +498,7 @@ def output(self):
             Cylinder surface.
         """
         self.Update()
-        output = wrap(self.GetOutput())
-        output.rotate_z(-90, inplace=True)
```

Comment:
I agree that we need to remove these  lines from this method.

We need to move these rotate and translate lines back into  `Cylinder` from `geometric_object.py`.  Unfortunately, pyvista did not use the default behavior from vtk for `Cylinder` and now it is inconsistent.  In my opinion, it is more preferable that `CylinderSource` is internally consistent.  We should document this difference in the documentation of both `Cylinder` and `CylinderSource`.

Reply:
```py
>>> import pyvista
>>> import numpy as np
>>> cylinder = pyvista.Cylinder(
...     center=[1, 2, 3], direction=[1, 1, 1], radius=1, height=2
... )
>>> cylinder.plot(show_edges=True, line_width=5, cpos='xy')
```

| Before | After |
| ------- | -----  |
| ![image](https://github.com/pyvista/pyvista/assets/70969910/77a65747-fc3e-4135-8d8f-32473c0f91b5) | ![image](https://github.com/pyvista/pyvista/assets/70969910/1e6a8f46-4e3e-489e-9205-752a1a73052f) |

is "After" the desired behavior?

Reply:
Just to clarify, the "Before" is when the rotate and translate is removed from the `pv.CylinderSource`

Reply:
@Arnav-2004 
Good point. Our goal is to adapt the behavior of the `Cylinder` function to the previous version. Could you please install the last released version with the following command to see if the display is the same?
```bash
pip install pyvista==0.42.2
```

Reply:
This is from pyvista==0.42.2
![image](https://github.com/pyvista/pyvista/assets/70969910/d0434a70-2373-4185-8849-0069c9ae9a3c)

Reply:
@Arnav-2004 Great. So, AFTER is the result that we want to have. Is there anything else you need us to go over?